### ISSUE TITLE
[feature request] Allow void return type for `@OnEditorAction`

### ISSUE THREAD

**Issue by BoD** [JakeWharton/butterknife#228]
When using a form, it is very frequent to want the "send" keyboard button to do the exact same thing as the action button of your UI.

As such, it would be very cool to be able to do:

    @OnEditorAction(R.id.edtPassword)
    @OnClick(R.id.btnLogin)
    protected void onLoginClicked() { ... }

But currently it is not possible because `@OnEditorAction` wants methods returning a boolean.
Maybe `void` could be allowed (and  `true` be assumed to be the default return value) in this case.

Another way would be to allow a `boolean` return value for `@OnClick` :)  But it's probably less desirable.

**Comment by JakeWharton** [comment]
Interesting use case. Is the send/done/whatever button the only thing that fires the editor action when the keyboard is up? I have always verified the action was what I was expecting.

**Comment by BoD** [comment]
Yes in my tests, this is fired only on the "send" button.  It is not fired for "next" for instance.

**Comment by dkozel** [comment]
I have the same use case which has lead to the following type of shim:

``` java
@OnEditorAction(R.id.mobileNumberField)
protected boolean mobileNumberFieldShim(int actionId) {
  if (actionId == EditorInfo.IME_ACTION_DONE) {
    signupUser();
  }

  return false;
}
```

It would be nice to be able to declare the `OnEditorAction` annotation with an argument specify what actionId to subscribe to.

``` java
@OnEditorAction(view = R.id.editTextField, action = EditorInfo.IME_ACTION_DONE)
protected void signupUser() { ... }
```

**Comment by JakeWharton** [comment]
Interesting proposal. It's not going to make 8.0, but I'll think about it for the future!

**Comment by rodrigohenriques** [comment]
I was thinking about something like that this week. Nice proposal.

**Comment by sosite** [comment]
Another example usage of above proposal:

```java
@OnEditorAction(value = R.id.password_input, action = EditorInfo.IME_ACTION_DONE)
@OnClick(R.id.signup_btn)
protected void signupUser() { ... }
```

**Comment by g-mac** [comment]
This would be a nice little tweak! I was actually looking for the same thing when I came across this. Is this still happening any time soon?
Thanks =) Simon
